py/modmicropython.c

```diff
@@ -80,13 +79,16 @@ mp_obj_t mp_micropython_mem_info(size_t n_args, const mp_obj_t *args) {
     #endif
     #if MICROPY_ENABLE_GC
     gc_dump_info();
+    #if MICROPY_PY_MICROPYTHON_MEM_INFO_BLOCKS || MICROPY_PY_MICROPYTHON_MEM_INFO_FRAGMENTATION
     if (n_args == 1) {
-        // arg given means dump gc allocation table
-        gc_dump_alloc_table();
+        mp_int_t arg = mp_obj_get_int(args[0]);
+        gc_dump_alloc_table(arg == 1, arg == 2);
```

Comment:
Did you intend this to be: `gc_dump_alloc_table(arg >= 1, arg >= 2);` ?  That would match the docs.

Or maybe it could be `gc_dump_alloc_table(arg & 1, arg & 2)`, which would allow better control over what's printed (sometimes you may only want the frag stats).

Reply:
Done. Also clarified the docs.